### PR TITLE
engineering: domain model in DDD

### PR DESCRIPTION
**Pull request by R-Jim**
#### What's this PR do?

- [x] Domain model in DDD

### CODE REVIEW

File: Engineering/Domain model in Domain Driven Design.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+tags: engineering, domain-driven-design, DDD
+author: M.Vu Cuong(Jim)
+---
+
+### Recap on Domain Driven Design
+
+A software design approach to create a rich and meaningful **Domain model** which enforces the conversation around the domain. As result, removes _the communication lag_ between **Developers** and **Domain/Business experts**.
+
+[Here is the full overview](obsidian://open?vault=brain&file=Engineering%2FOverview%20of%20Domain%20Driven%20Design)
```
Comment (zlatanpham) [inline Engineering/Domain model in Domain Driven Design.md:10]: I think including the recap is kind of unnecessary. Maybe just mention the DDD article from the introduction of Domain model. e.g:

```
In [Domain Driven Design](link), Domain Model is ....
```

File: Engineering/Domain model in Domain Driven Design.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+The vocabulary and key concepts of the domain, and the relationships among all of the entities.
+These are mainly divided into: **Domain events**, **Commands**, **Aggregates**, and **Bounded context**.
+### Domain events
+They are factual statements about the **things that happened** in a business system. It is important to phrase these statements in the past tense so one can frame this as a ‘what happened’ statement.
+Exp: Order submitted, Shopping cart updated
+### Commands
+Is the identifier to why the **domain events** occurred. Commands express our intent for something to happen in the future as used in present tense. Commands may be documented as both user and system actions.
+Exp: **Add product(Command)** -> Shopping cart updated(Domain event)
+### Aggregrates
+It is represented by a cluster of events with corresponding commands and the responsible actor.
```
Comment (monotykamary) [inline Engineering/Domain model in Domain Driven Design.md:24]: This statement is not unique and is taken directly from the reference source:
- https://creately.com/blog/diagrams/event-storming/

Please paraphrase this statement.

**Review by monotykamary — CHANGES_REQUESTED** [review]

File: Engineering/Domain model in Domain Driven Design.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+---
+tags: engineering, domain-driven-design, DDD
+author: M.Vu Cuong(Jim)
+---
+### Domain model is
```
Comment (monotykamary) [inline Engineering/Domain model in Domain Driven Design.md:5]: https://github.com/dwarvesf/brain/blob/master/CONTRIBUTING.md
Last nit picks; use ## H2 headers as your top level header.

File: Engineering/Domain model in Domain Driven Design.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+Your organized and structured **knowledge of the business problem**. Represents as a diagram, code examples, or written documentation, and **must** be accessible and understandable by **everyone** involved with the project.
+[Overview of Domain Driven Design](https://github.com/dwarvesf/brain/blob/master/Engineering/Overview%20of%20Domain%20Driven%20Design.md)
+### Domain model consists of
```
Comment (monotykamary) [inline Engineering/Domain model in Domain Driven Design.md:8]: Likewise, use ## H2 headers to mirror your top level header.

File: Engineering/Domain model in Domain Driven Design.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+The vocabulary and key concepts of the domain, and the relationships among all of the entities.
+These are mainly divided into: **Domain events**, **Commands**, **Aggregates**, and **Bounded context**.
+### Domain events
```
Comment (monotykamary) [inline Engineering/Domain model in Domain Driven Design.md:11]: Likewise, use ## H2 headers to mirror your top level header.

File: Engineering/Domain model in Domain Driven Design.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+*A statement in past tense* describes the **things that happened** in a business system that alternates the state of the entity.
+Exp: Order submitted, Shopping cart updated
+### Commands
```
Comment (monotykamary) [inline Engineering/Domain model in Domain Driven Design.md:14]: Likewise, use ## H2 headers to mirror your top level header.

File: Engineering/Domain model in Domain Driven Design.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+*A verb in present tense* describes the action that triggers the corresponding **domain event**. It is either user or system actions.
+- Exp: `Add product`(Command) -> `Shopping cart updated`(Domain event)
+### Aggregrates
```
Comment (monotykamary) [inline Engineering/Domain model in Domain Driven Design.md:17]: Likewise, use ## H2 headers to mirror your top level header.

File: Engineering/Domain model in Domain Driven Design.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+Represented by a *minimal cluster* of associated objects(domain events, commands, and actors) that we treat as a unit for data change. Each has a boundary and only exposes its root(**Aggregate Root**) which allows other objects to reference it.
+- Exp: A team wants to update its member role according to each project: </n>
+`Project, Update member role` (Aggregate Root) -> `Project's member role updated`</n>
+### Bounded context
```
Comment (monotykamary) [inline Engineering/Domain model in Domain Driven Design.md:21]: Likewise, use ## H2 headers to mirror your top level header.

File: Engineering/Domain model in Domain Driven Design.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+A high-level structure consists of categorizations of functionality, represents a circle or square, that groups related entities together. It can bound parts of an aggregate or multiple aggregates.
+- Exp: In an aggregate for the shopping process,  we draw the bounded contexts for **Shopping cart**, and **Offers**: </n>
+Shopping cart(`User` -> `Add product to cart` -> `Cart updated`) -> Offers(`Promotiational Offers Identified` -> `Offers added`)
+### References
```
Comment (monotykamary) [inline Engineering/Domain model in Domain Driven Design.md:25]: Likewise, use ## H2 headers to mirror your top level header.

**Review by monotykamary — CHANGES_REQUESTED** [review]
Just needs some work on header styling,

otherwise LGTM.

File: Engineering/Domain model in Domain Driven Design.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
+---
+tags: engineering, domain-driven-design, DDD
+author: M.Vu Cuong(Jim)
+---
+## Domain model is
```
Comment (zlatanpham) [inline Engineering/Domain model in Domain Driven Design.md:5]:
```suggestion
## What is Domain Model
```

File: Engineering/Domain model in Domain Driven Design.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+Your organized and structured **knowledge of the business problem**. Represents as a diagram, code examples, or written documentation, and **must** be accessible and understandable by **everyone** involved with the project.
```
Comment (zlatanpham) [inline Engineering/Domain model in Domain Driven Design.md:6]:
```suggestion
In [Overview of Domain Driven Design | Domain Driven Design], Domain Model is the organized and structured **knowledge of the business problem**. Represents as a diagram, code examples, or written documentation, and **must** be accessible and understandable by **everyone** involved with the project.
```

File: Engineering/Domain model in Domain Driven Design.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+[Overview of Domain Driven Design](https://github.com/dwarvesf/brain/blob/master/Engineering/Overview%20of%20Domain%20Driven%20Design.md)
```
Comment (zlatanpham) [inline Engineering/Domain model in Domain Driven Design.md:7]: Remove this

File: Engineering/Domain model in Domain Driven Design.md [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+## Domain model consists of
```
Comment (zlatanpham) [inline Engineering/Domain model in Domain Driven Design.md:8]:
```suggestion
## Components of Domain Model
```

File: Engineering/Domain model in Domain Driven Design.md [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+The vocabulary and key concepts of the domain, and the relationships among all of the entities.
+These are mainly divided into: **Domain events**, **Commands**, **Aggregates**, and **Bounded context**.
```
Comment (zlatanpham) [inline Engineering/Domain model in Domain Driven Design.md:10]:
```suggestion
The vocabulary and key concepts of the domain, and the relationships among all of the entities. These are mainly divided into: **Domain events**, **Commands**, **Aggregates**, and **Bounded context**.
```

File: Engineering/Domain model in Domain Driven Design.md [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+## Domain events
```
Comment (zlatanpham) [inline Engineering/Domain model in Domain Driven Design.md:11]:
```suggestion
### Domain events
```

File: Engineering/Domain model in Domain Driven Design.md [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+*A statement in past tense* describes the **things that happened** in a business system that alternates the state of the entity.
+Exp: Order submitted, Shopping cart updated
+## Commands
```
Comment (zlatanpham) [inline Engineering/Domain model in Domain Driven Design.md:14]:
```suggestion
### Commands
```

File: Engineering/Domain model in Domain Driven Design.md [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+*A verb in present tense* describes the action that triggers the corresponding **domain event**. It is either user or system actions.
+- Exp: `Add product`(Command) -> `Shopping cart updated`(Domain event)
+## Aggregrates
```
Comment (zlatanpham) [inline Engineering/Domain model in Domain Driven Design.md:17]:
```suggestion
### Aggregrates
```

File: Engineering/Domain model in Domain Driven Design.md [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+Represented by a *minimal cluster* of associated objects(domain events, commands, and actors) that we treat as a unit for data change. Each has a boundary and only exposes its root(**Aggregate Root**) which allows other objects to reference it.
+- Exp: A team wants to update its member role according to each project: </n>
+`Project, Update member role` (Aggregate Root) -> `Project's member role updated`</n>
+## Bounded context
```
Comment (zlatanpham) [inline Engineering/Domain model in Domain Driven Design.md:21]:
```suggestion
### Bounded context
```

**Review by zlatanpham — CHANGES_REQUESTED** [review]
I've left some suggestions to make the note better formatted.

**Review by zlatanpham — APPROVED** [review]

**Pull request merged**